pkg/instrumentation/sdk.go

```diff
@@ -488,22 +535,29 @@ func (i *sdkInjector) createResourceMap(ctx context.Context, otelinst v1alpha1.I
 	k8sResources[semconv.K8SPodNameKey] = pod.Name
 	k8sResources[semconv.K8SPodUIDKey] = string(pod.UID)
 	k8sResources[semconv.K8SNodeNameKey] = pod.Spec.NodeName
-	k8sResources[semconv.ServiceInstanceIDKey] = createServiceInstanceId(ns.Name, fmt.Sprintf("$(%s)", constants.EnvPodName), pod.Spec.Containers[index].Name)
+	k8sResources[semconv.ServiceInstanceIDKey] = createServiceInstanceId(pod, useLabelsForResourceAttributes, ns.Name, fmt.Sprintf("$(%s)", constants.EnvPodName), pod.Spec.Containers[index].Name)
 	i.addParentResourceLabels(ctx, otelinst.Spec.Resource.AddK8sUIDAttributes, ns, pod.ObjectMeta, k8sResources)
+
 	for k, v := range k8sResources {
 		if !existingRes[string(k)] && v != "" {
 			res[string(k)] = v
 		}
 	}
 
-	for k, v := range pod.GetAnnotations() {
-		if strings.HasPrefix(k, constants.OtelAnnotationNamespace) {
-			key := strings.TrimSpace(strings.TrimPrefix(k, constants.OtelAnnotationNamespace))
-			if _, ok := res[key]; !ok {
-				res[key] = v
+	// attributes and labels from the pod have the highest precedence (except for values set in environment variables)
+	for k, v := range pod.Annotations {
+		if strings.HasPrefix(k, constants.ResourceAttributeAnnotationPrefix) {
+			k = strings.TrimPrefix(k, constants.ResourceAttributeAnnotationPrefix)
+			if !existingRes[k] && k != string(semconv.ServiceNameKey) {
```

Comment:
don't we want to let the override happen here?

Reply:
we treat the user defined enviroment variables as the highest priority - otherwise it would be a breaking change.
Not sure if it would also be surprising.

https://github.com/zeitlinger/opentelemetry-operator/blob/c1966659f203516c87991715c95f1dc00a372aa1/README.md?plain=1#L783